### ISSUE TITLE
Clarify naming

### ISSUE THREAD

**Issue by mjg** [mjsottile/sfsexp#14]
Before possibly packaging this library, it would be nice to know the preferred name. So far I see:

- **sfsexp**: GitHub project, GitHub source archive name, package config `sfsexp.pc`
- **sexp**: header, shared and static libraries, python module and setup.py; by far the most used term in the source tree
- **sexpr**: SourceForge project, win32 vcproject (and library?); lib which `pysexp.spec` references; release tarball

And I honestly don't know how to name a package which builds and installs `libsexp.so` and `sfsexp.pc` from `sexpr-1.3.1.tar.gz` ... None of this is a big problem, but it would help users of your library if packages in all distros have the same name - the choice should be yours!

**Comment by mjg** [comment]
Two data points:
- [Debian](https://sources.debian.org/src/sfsexp/) and [derivatives](https://repology.org/project/sfsexp/versions) use sfsexp
- [ALT Linux](http://www.sisyphus.ru/en/srpm/Sisyphus/sexpr) uses sexpr, see also [repology](https://repology.org/project/sexpr/versions)

Truth in advertising: Fedora's notmuch packager here, willing to package sfsexp/sexpr in Fedora (for notmuch purposes).

**Comment by mjsottile** [comment]
Thanks for the note.  Yes, I agree - I haven't been very consistent over the lifetime of the project in terms of naming.  Of the three names, only the first two matter (sfsexp and sexp).  The third one is mostly legacy stuff: 

- The SF.net repository/site is not used any longer and only exists to make sure old web links can redirect to the new home at github.  It can be disregarded for all intents and purposes.
- The win32, python, ruby, and SILOON components can all be ignored - they aren't maintained and haven't been tested for many years (in some cases they haven't been touched in almost 20 years).  I really should just remove them.  I'll create an issue so I don't forget to do some repository tidying at some point.

So that brings us to the two remaining cases: sexp vs sfsexp.  The rationale for naming the project sfsexp was that sexp is not exactly unique, so it would be confusing to call it that since it would be impossible to search for.  sfsexp on the other hand is unique, so if you search for it you can find this library.  On the other hand, the headers and data structures use the sexp name since they refer to the generic concept of an s-expression, not the project name.  The library itself - that should probably have been libsfsexp.so since that refers to the project instead of the generic structure.  Unfortunately, I don't like breaking existing things so renaming it might be hard - plus, I think the name "sfsexp" was created in the last 10 years, while the makefile and codebase itself is over 20 years old.  The naming of the files and libraries got a bit calcified before the code itself was separated from its parent project and released as a standalone project around 2003.

In short: the package should be called sfsexp.  The name sexpr is legacy from the old SF.net project and should be avoided.

**Comment by mjsottile** [comment]
@mjg Thanks for the fixes!  Merged.
